sdk/go/common/testing/diagtest/sink.go

```diff
@@ -39,3 +40,26 @@ func LogSink(t testing.TB) diag.Sink {
 		},
 	)
 }
+
+// Wraps a testing.TB and intercepts log messages.
+type FakeT struct {
```

Comment:
Rather than exposing this and using diagtest.LogSink(&fakeT) in the tests just write a new `MockSink` in sdk/go/common/testing/diagtest/sink.go that just writes to a io.Writer and use a bytes.Buffer to receive that in the tests. Then you can just check the overall text stream contains the snippet you want.

Reply:
updated